services-custom/dynamodb-enhanced/README.md

```diff
@@ -253,6 +254,85 @@ how Lombok's 'onMethod' feature is leveraged to copy the attribute based DynamoD
     }
 ```
 
+### Using subtypes to assist with single-table design
+It's considered a best practice in some situations to combine entities of various types into a single table in DynamoDb
+to enable the querying of multiple related entities without the need to actually join data across multiple tables. The
+enhanced client assists with this by supporting polymorphic mapping into distinct subtypes.
+
+Let's say you have a customer:
+
+```java
+public class Customer {
+    String getCustomerId();
+    void setId(String id);
+    
+    String getName();
+    void setName(String name);
+}
+```
+
+And an order that's associated with a customer:
+
+```java
+public class Order {
+    String getOrderId();
+    void setOrderId();
+    
+    String getCustomerId();
+    void setCustomerId();
+}
+```
+
+You could choose to store both of these in a single table that is indexed by customer ID, and create a TableSchema that 
+is capable of mapping both types of entities into a common supertype:
+
+```java
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSubtypeName;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSubtypes;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSubtypes.Subtype;
+
+@DynamoDbBean
+@DynamoDbSubtypes({
+        @Subtype(name = "CUSTOMER", subtypeClass = Customer.class),
+        @Subtype(name = "ORDER", subtypeClass = Order.class)})
+public class CustomerRelatedEntity {
```

Comment:
Java code would be way neater if this could be an interface and classes implementing it. Is that a feasible option?

Reply:
@dvlato could you elaborate on exactly what you are proposing? Maybe with an example?